### ISSUE TITLE
Linux: crash on startup if window is not in the foreground

### ISSUE THREAD

**Issue by jwise** [bambulab/BambuStudio#1149]
If I alt-tab away from Bambu Studio while I am launching it at just the wrong time, the slicer crashes.  The backtrace is:

```
#0  0x0000000000000000 in  ()
#1  0x0000555559f9bdc5 in Slic3r::GUI::GLModel::send_to_gpu(Slic3r::GUI::GLModel::RenderData&, std::vector<float, std::allocator<float> > const&, std::vector<unsigned int, std::allocator<unsigned int> > const&)
    (this=<optimized out>, data=..., vertices=std::vector of length 900, capacity 900 = {...}, indices=std::vector of length 150, capacity 150 = {...})
    at /home/joshua/work/BambuStudio/src/slic3r/GUI/GLModel.cpp:315
#2  0x0000555559f9f33d in Slic3r::GUI::GLModel::init_from(indexed_triangle_set const&, Slic3r::BoundingBoxf3 const&) (this=<optimized out>, its=..., bbox=<optimized out>)
    at /home/joshua/work/BambuStudio/src/slic3r/GUI/GLModel.cpp:106
#3  0x0000555559fa2756 in Slic3r::GUI::GLModel::init_from(indexed_triangle_set const&) (this=this@entry=0x618000252550, its=...)
    at /home/joshua/work/BambuStudio/src/slic3r/GUI/GLModel.cpp:112
#4  0x0000555559bc65b4 in Slic3r::GUI::GLGizmoBase::GLGizmoBase(Slic3r::GUI::GLCanvas3D&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int) (this=this@entry=0x618000252480, parent=<optimized out>, icon_filename=<optimized out>, sprite_id=<optimized out>)
    at /home/joshua/work/BambuStudio/src/slic3r/GUI/Gizmos/GLGizmoBase.cpp:164
#5  0x0000555559bcb93f in Slic3r::GUI::GLGizmoMove3D::GLGizmoMove3D(Slic3r::GUI::GLCanvas3D&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, Slic3r::GUI::GizmoObjectManipulation*)
    (this=this@entry=0x618000252480, parent=<optimized out>, icon_filename="toolbar_move.svg", sprite_id=sprite_id@entry=0, obj_manipulation=obj_manipulation@entry=0x62500070fcb0) at /home/joshua/work/BambuStudio/src/slic3r/GUI/Gizmos/GLGizmoMove.cpp:32
#6  0x0000555559b7ae36 in Slic3r::GUI::GLGizmosManager::init() (this=this@entry=0x62500070faa8) at /home/joshua/work/BambuStudio/src/slic3r/GUI/Gizmos/GLGizmosManager.cpp:189
#7  0x000055555995c994 in Slic3r::GUI::GLCanvas3D::init() (this=<optimized out>) at /home/joshua/work/BambuStudio/src/slic3r/GUI/GLCanvas3D.cpp:1220
#8  0x000055555a2c7457 in Slic3r::GUI::GUI_App::post_init() (this=<optimized out>) at /home/joshua/work/BambuStudio/src/slic3r/GUI/GUI_App.cpp:1025
```

The root cause is that `::glGenBuffers` (`__glewGenBuffers`) is NULL, presumably because `init_opengl` failed (and was not error checked).

Two things to do:
* error-check `init_opengl`, and if it fails, pop up a message box and then croak
* figure out why `init_opengl` fails if the window is not in the foreground, and how to avoid this

**Comment by jwise** [comment]
Proposed fix in #1198.

**Comment by jwise** [comment]
Closed by #1198 .
